java/ql/lib/semmle/code/java/security/SensitiveKeyboardCacheQuery.qll

```diff
@@ -0,0 +1,53 @@
+/** Definitions for the keyboard cache query */
+
+import java
+import semmle.code.java.security.SensitiveActions
+
+/** An Android Layout XML file. */
+class AndroidLayoutXmlFile extends XmlFile {
+  AndroidLayoutXmlFile() { this.getAbsolutePath().matches("%/res/layout/%.xml") }
+}
+
+/** An XML element that represents an editable text field. */
+class AndroidEditableXmlElement extends XmlElement {
+  XmlAttribute inputType;
+  XmlAttribute id;
```

Comment:
Suggest using the `AndroidXmlAttribute` type so that you can remove the namespace prefix checks below:

```suggestion
  AndroidXmlAttribute inputType;
  AndroidXmlAttribute id;
```